### DISCUSSION TITLE
Vector.Ceilling<T>, Vector.Floor<T>

### DISCUSSION THREAD

**Discussion by giladfrid009** [dotnet/runtime#62991] [Ideas] [answered]
In System.Numerics.Vector static class, in .NET6, the following ceiling and floor types are supported:

- ``System.Numerics.Vector.Ceiling(Vector<Double>)``
- ``System.Numerics.Vector.Ceiling(Vector<Single>)``

Since the class Vector<T> is generic, is there a reason there are no generic implementation for these methods? 
Especially since Ceiling and Floor are both well defined for floating point and integer types.
Also, a generic method would allow cleaner future extension for other types like Half - only one method required for all floating types, instead of separate methods for each individual type.

Demonstration of the desired method:

```csharp
public static Vector<T> System.Numerics.Vector.Ceiling<T>(Vector<T> vector)
{
       //All 'typeof' checks are optimized in compile time by JIT, 
       //Also could possibly be implemented nicely with Generic Math in the future (static abstracts in interfaces)

	if (typeof(T) == typeof(Double))
	{
		return Vector.Ceiling(vector); // internal Double implementation
	}
	if (typeof(T) == typeof(Single))
	{
		return Vector.Ceiling(vector); //internal Single implementation
	}
        //possible future ceiling support for half and other floating point types
        //if (typeof(T) == typeof(Half)) 
	//{
	//	return Vector.Ceiling(vector); //internal Half implementation
	//}

	return vector; // ceiling does nothing
}
```

**Comment by huoyaoyuan** [comment, marked as answer]
Sounds somehow reasonable. You can open an API suggestion issue.
